src/core/lib/iomgr/tcp_posix.cc

```diff
@@ -1011,30 +1009,30 @@ static bool tcp_do_read(grpc_tcp* tcp, grpc_error_handle* error)
 static void maybe_make_read_slices(grpc_tcp* tcp)
     ABSL_EXCLUSIVE_LOCKS_REQUIRED(tcp->read_mu) {
   if (tcp->incoming_buffer->length <
-          static_cast<size_t>(tcp->min_progress_size) &&
-      tcp->incoming_buffer->count < MAX_READ_IOVEC) {
-    if (GRPC_TRACE_FLAG_ENABLED(grpc_tcp_trace)) {
-      gpr_log(GPR_INFO,
-              "TCP:%p alloc_slices; min_chunk=%d max_chunk=%d target=%lf "
-              "buf_len=%" PRIdPTR,
-              tcp, tcp->min_read_chunk_size, tcp->max_read_chunk_size,
-              tcp->target_length, tcp->incoming_buffer->length);
+      static_cast<size_t>(tcp->min_progress_size)) {
+    const size_t max_read_iovec =
```

Comment:
Could you add a comment on why we're controlling max_read_iovec like this

Reply:
done